numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,35 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape=None):
+    """
+    Create rolling window views of the 2D array with the given shape.
```

Comment:
No reason to limit to 2D

Reply:
I know. I plan to implement beyond 2d and strides > 2 soon.

Reply:
It looks like what you have already works fine for > 2

Reply:
Actually not. 

This is a 3D test case:
```
arr = np.arange(27).reshape(3,3,3)
print(arr)
shape = [2,2,2]
sliding_window_view(arr, shape)

input:
[[[ 0  1  2]
  [ 3  4  5]
  [ 6  7  8]]

 [[ 9 10 11]
  [12 13 14]
  [15 16 17]]

 [[18 19 20]
  [21 22 23]
  [24 25 26]]]

output:
array([[[[[[ 0,  1],
           [ 3,  4]],

          [[ 9, 10],
           [12, 13]]],


         [[[ 1,  2],
           [ 4,  5]],

          [[10, 11],
           [13, 14]]]],



        [[[[ 3,  4],
           [ 6,  7]],

          [[12, 13],
           [15, 16]]],


         [[[ 4,  5],
           [ 7,  8]],

          [[13, 14],
           [16, 17]]]]],




       [[[[[ 9, 10],
           [12, 13]],

          [[18, 19],
           [21, 22]]],


         [[[10, 11],
           [13, 14]],

          [[19, 20],
           [22, 23]]]],



        [[[[12, 13],
           [15, 16]],

          [[21, 22],
           [24, 25]]],


         [[[13, 14],
           [16, 17]],

          [[22, 23],
           [25, 26]]]]]])
```

You can see there is some repeated elements. The expected output dims should be (2, 2, 2)

Reply:
That output looks correct to me - `(2, 2, 2, 2, 2, 2)` is the right shape for a sliding window of shape `(2, 2, 2)` over an array of shape `(3, 3, 3)`

Reply:
It should work on N(>3) dim.